components/function-controller/pkg/apis/serverless/v1alpha1/function_defaults.go

```diff
@@ -0,0 +1,61 @@
+package v1alpha1
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"knative.dev/pkg/webhook/resourcesemantics"
+)
+
+var _ resourcesemantics.GenericCRD = (*Function)(nil)
+
+func (fn *Function) SetDefaults(_ context.Context) {
+	fn.Spec.defaultResources()
+	fn.Spec.defaultReplicas()
+}
+
+func (spec *FunctionSpec) defaultReplicas() {
+	if spec.MinReplicas == nil {
+		one := int32(1)
+		spec.MinReplicas = &one
+	}
+	if spec.MaxReplicas == nil {
+		newMax := int32(1)
+		if *spec.MinReplicas > newMax {
```

Comment:
What if user pass only `maxReplicas: 0`? :) Then he/she will have `min: 1` and `max: 1`. If user pass only max, then we should default min with value from max.